editor/src/clj/editor/code/data.clj

```diff
@@ -1911,7 +1911,11 @@
               (assoc :regions regions')))))
 
 (defn- begins-indentation? [grammar ^String line]
-  (and (some? line) (some? (some-> grammar :indent :begin (re-find line)))))
+  (when (some? line)
+    (let [begin? (:begin (:indent grammar))]
+      (cond
+        (fn? begin?) (begin? line)
+        :else (re-find begin? line)))))
```

Comment:
This is an experimental fix. I know it's not very elegant, but I'm not sure how to avoid using this complex regexp another way while keeping the functionality. I'm open to suggestions.